src/tests/change.spec.ts

```diff
@@ -0,0 +1,155 @@
+    class UserAdmin extends AdminEntity {
+      entity = User
+    }
+
+    @Module({
+      imports: [TypeOrmModule.forFeature([User]), TestAuthModule, DefaultCoreModule],
+      exports: [TypeOrmModule],
+    })
+    class RegisteredAdminEntityModule {
+      constructor(private readonly adminSite: DefaultAdminSite) {
+        adminSite.register('user', UserAdmin)
+      }
+    }
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TestTypeOrmModule.forRoot({ entities: [User, Agency, Group] }),
+        RegisteredAdminEntityModule,
+      ],
+    }).compile()
+
+    const app = module.createNestApplication()
+    await app.init()
+
+    const userData = createTestUser({ firstName: 'Pallavi' })
+    const userRepository: Repository<User> = app.get(getRepositoryToken(User))
+    const user = await userRepository.save(userData)
+    expect(await userRepository.findOneOrFail(user.id)).toBeDefined()
+
+    const server = app.getHttpServer()
+    const req = await request(server).get(`/admin/user/user/${user.id}/change`)
+
+    document.documentElement.innerHTML = req.text
+
+    expect(document.querySelector('label[for="firstName"]')).toBeTruthy()
+    expect(document.querySelector('label[for="lastName"]')).toBeTruthy()
+    expect(document.querySelector('label[for="gender"]')).toBeTruthy()
+    expect(document.querySelector('label[for="createdDate"]')).toBeTruthy()
+    expect(document.querySelector('label[for="password"]')).toBeTruthy()
+    expect(document.querySelector('label[for="email"]')).toBeTruthy()
+    expect(document.querySelector('label[for="roles"]')).toBeTruthy()
+    expect(document.querySelector('label[for="moreRoles"]')).toBeTruthy()
+    expect(document.querySelector('label[for="description"]')).toBeTruthy()
+    expect(document.querySelector('label[for="agency"]')).toBeTruthy()
+    expect(document.querySelector('label[for="groups"]')).toBeTruthy()
+    expect(document.querySelector('label[for="age"]')).toBeTruthy()
+    expect(document.querySelector('label[for="numberOfSiblings"]')).toBeTruthy()
+    expect(document.querySelector('label[for="fingerCount"]')).toBeTruthy()
+    expect(document.querySelector('label[for="atomCount"]')).toBeTruthy()
+    expect(document.querySelector('label[for="height"]')).toBeTruthy()
+    expect(document.querySelector('label[for="bmi"]')).toBeTruthy()
+    expect(document.querySelector('label[for="birthdate"]')).toBeTruthy()
+    expect(document.querySelector('label[for="deathdate"]')).toBeTruthy()
+    expect(document.querySelector('label[for="birthtime"]')).toBeTruthy()
+    expect(document.querySelector('label[for="deathtime"]')).toBeTruthy()
+    expect(document.querySelector('label[for="hireDatetime"]')).toBeTruthy()
+    expect(document.querySelector('label[for="isActive"]')).toBeTruthy()
+    expect(document.querySelector('label[for="additionalData"]')).toBeTruthy()
+    expect(document.querySelector('label[for="updatedDate"]')).toBeTruthy()
```

Comment:
A better, less brittle test would be something like:

```ts
for (let property in user) {
     expect(document.querySelector(`label[for="${property}"]`)).toBeTruthy();
}
```

It's also more future-proof: if we add properties to the `User`, they will be tested